File: app/src/main/java/cn/guluwa/gulumusic/listener/OnSongStatusListener.java
package cn.guluwa.gulumusic.listener;

import cn.guluwa.gulumusic.data.bean.TracksBean;

/**
 * Created by guluwa on 2018/2/2.
 */

public interface OnSongStatusListener {

    /**
     * 加载
     */
    void loading();

    /**
     * 开始
     */
    void start();

    /**
     * 暂停
     */
    void pause();

    /**
     * 结束
     *
     * @param tracksBean
     */
    void end(TracksBean tracksBean);

    /**
     * 错误
     *
     * @param msg
     */
    void error(String msg);

    /**
     * 进度
     *
     * @param progress
     */
    void progress(int progress, int duration);
}

File: app/src/main/java/cn/guluwa/gulumusic/listener/OnSongStatusListener.kt
package cn.guluwa.gulumusic.listener

import cn.guluwa.gulumusic.data.bean.TracksBean

/**
 * Created by guluwa on 2018/2/2.
 */

interface OnSongStatusListener {

    /**
     * 加载
     */
    fun loading()

    /**
     * 开始
     */
    fun start()

    /**
     * 暂停
     */
    fun pause()

    /**
     * 结束
     *
     * @param tracksBean
     */
    fun end(tracksBean: TracksBean)

    /**
     * 错误
     *
     * @param msg
     */
    fun error(msg: String)

    /**
     * 进度
     *
     * @param progress
     */
    fun progress(progress: Int, duration: Int)

    /**
     * 歌曲封面图
     *
     * @param url
     */
    fun pic(url: String)

    /**
     * 歌曲缓存完成
     */
    fun download(position:Int)
}
